### ISSUE TITLE
[Feature Request] Enable nn.Module with Python copy

### ISSUE THREAD

**Issue by 0x1orz** [ml-explore/mlx#300]
To reuse a  module copy,  like:
```python
from copy import deepcopy
import  mlx.nn as mlx_nn
repeat = lambda module, n: [deepcopy(module) for _ in range(n)]

mlx_layer = mlx_nn.Linear(2, 3)
mlx_layers = repeat(mlx_layer, 2)
```
that  raised `TypeError: cannot pickle 'mlx.core.array' object` in `deepcopy(module)`.

**Comment by ekdnam** [comment]
i would like to work on this!

**Comment by awni** [comment]
Great! If you have any questions about it, we can use this issue for discussion or better yet start a draft PR and use that!

**Comment by ekdnam** [comment]
Yes, I will start working on it! Could you point to where should I look into the code to start?

**Comment by NripeshN** [comment]
Hey @awni,
Could you please ensure contributors link the PR to the issue properly before merging(eg closes <issue>). This would automatically close issues.

**Comment by awni** [comment]
Good point! Fixed in #335

**Comment by awni** [comment]
Spoke too soon. This is not actually fixed (se #366 )
